### PR TITLE
Upgraded node and react version / Implemented ActionMenu

### PR DESCRIPTION
**Pull request by makhmudjon-inadullaev**
Updated react's version and `ActionMenu` implemented
Update for https://github.com/github/accessibility-audits/issues/2748#issuecomment-1298500415.

## References

Comment: https://github.com/github/accessibility-audits/issues/2748#issuecomment-1298500415
https://github.com/github/accessibility-audits/issues/2797
https://github.com/github/accessibility-audits/issues/2796
https://github.com/github/accessibility-audits/issues/2794
https://github.com/github/accessibility-audits/issues/2748
https://github.com/github/accessibility-audits/issues/2798

### CODE REVIEW

**Comment by darcyclarke** [comment]
@makhmudjon-inadullaev / @pavlov-kirill can you please wait to push/merge updates prior to review from the @npm/cli-team? We've assumed primary responsibility over the repo/theme prior to this accessibility work & just want to ensure clear communication about sign-off/reviews

**Comment by narula0781** [comment]
@darcyclarke - Anybody from your team who we should add for the reviews?
We would need your support in quickly closing the reviews as we already have multiple level of reviews by current npm registry team , accessibility team. Thanks

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
       - name: Use Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: 16.x
+          node-version: 18.x
```
Comment (amit-avit) [inline .github/workflows/publish.yml:34]: Why is this needed?
Shouldn't node v16 be enough ?

**Comment by Seryozha95** [comment]
@pavlov-kirill can we add the selected property? Opened action menu should show the selected item. https://primer.style/react/ActionMenu#minimal-example

File: .github/workflows/publish.yml (hunk 1 above)
Comment (amit-avit) [reply .github/workflows/publish.yml]: Unresolving as it is not answered

**Comment by pavlov-kirill** [comment]
> @pavlov-kirill can we add the selected property? Opened action menu should show the selected item. https://primer.style/react/ActionMenu#minimal-example

@Seryozha95 added

Uploading Screen Recording 2022-11-09 at 18.41.09.mov…

**Review by Seryozha95 — APPROVED** [review]

**Comment by Seryozha95** [comment]
@npm/cli-team could yu please review this?

cc @darcyclarke

**Review by 20shivangi — COMMENTED** [review]
When I open the dropdown and suppose the version v7.24 is already selected, then the dropdown will be as shown below. So, as you can see both the selected version and the version which we are navigating via tab both shown with grey background.

Screenshot 2022-11-14 at 4 58 18 PM


Should be show the grey background only on version where tab is being navigated and selected version in bold only. Or should we keep it this way only, any suggestions ? @amit-avit @prabum3

**Comment by prabum3** [comment]
> Should be show the grey background only on version where tab is being navigated and selected version in bold only. Or should we keep it this way only, any suggestions 

I feel the current highlighting looks good. The blue bar clearly differentiates the chosen version and the version which is under focus

File: theme/src/components/variant-select.js [hunk 2]
```diff
@@ -68,48 +37,32 @@ function VariantSelect(props) {
...
-    if (event.key === 'Escape') {
-        document.body.click()
-    }
-  }
 
   return (
-    <div style={{ display: 'flex', flexDirection: 'column' }} ref={wrapper}>
+    <ThemeProvider>
       <label id="label-versions-list-item" htmlFor='variant-select'>Select CLI Version:</label>
-      <Dropdown id="versions-list-item" aria-labelledby={selectedItem?.variant?.shortName} onKeyDown={onDropDownKeyDown} overlay={props.overlay}>
```
Comment (prabum3) [inline theme/src/components/variant-select.js:105]: With the latest changes, VoiceOver on Mac does not announce the selected version (before clicking on the menu button). Can we fix it ?

Comment (pavlov-kirill) [reply theme/src/components/variant-select.js:105]: @prabum3 I added title for selectedItem to aria-label.

Comment (prabum3) [reply theme/src/components/variant-select.js:105]: It would be better to add aria-label to `ActionMenu` instead of `ActionMenu.Button`. Now the screen reader is announcing it twice when selected.

Comment (pavlov-kirill) [reply theme/src/components/variant-select.js:105]: @prabum3 I added condition for exposing `aria-label` only if `ActionMenu` closed. If add `aria-label` to ActionMenu it could be announce twice also.

**Review by prabum3 — APPROVED** [review]
🚀

**Comment by Seryozha95** [comment]
@npm/cli-team please review this pr
cc @darcyclarke cc @narula0781

**Review by darcyclarke — APPROVED** [review]
✅ LGTM

**Comment by darcyclarke** [comment]
@narula0781 `@lukekarrys` has been our DRI for keeping the docs/theme & CI up-to-date/working for the last several months & I know was in the process of trying to get it on-par/aligned with `github/docs` so that we could alleviate some maintenance burden. With that in mind, I hope you're working closely with that team to upstream any accessibility changes that are relevant to them as well.

File: theme/src/components/variant-select.js [hunk 3]
```diff
@@ -68,48 +37,33 @@ function VariantSelect(props) {
...
-          href={match.page.url}
-          onClick={e => anchorClickHandler(e, match.page.url)}
-          onKeyDown={e => onItemEnterKey(e, match.page.url)}
-          onBlur={index === (variantPages.length - 1) ? collapseDropdown : undefined}
-          aria-label={`${match.variant.title}. List items ${index + 1} of ${variantPages.length}`}
-        >
-          <Dropdown.Item>{match.variant.title}</Dropdown.Item>
-        </a>
-      );
+      items.push(<ActionList.Item onKeyDown={(e) => onItemEnterKey(e, match.page.url)} onClick={e => anchorClickHandler(e, match.page.url)} id={match.variant.shortName} key={index} active={active}>
```
Comment (amit-avit) [inline theme/src/components/variant-select.js:45]: nit

```suggestion
      items.push(<ActionList.Item
        onKeyDown={e => onItemEnterKey(e, match.page.url)}
        onClick={e => anchorClickHandler(e, match.page.url)}
        id={match.variant.shortName}
        key={index}
        active={active}>
```

**Review by amit-avit — APPROVED** [review]
This looks and works so much better.

**Review by 20shivangi — APPROVED** [review]
